discovery-infra/start_discovery.py

```diff
@@ -204,22 +204,22 @@ def create_nodes_and_wait_till_registered(
     )
 
     # TODO: Check for only new nodes
-    utils.wait_till_nodes_are_ready(
-        nodes_count=nodes_count, network_name=nodes_details["libvirt_network_name"]
-    )
     if not inventory_client:
+        utils.wait_till_nodes_are_ready(
```

Comment:
The log doesn't match the action. Why in case no inventory_url you want to wait for them to register?

Reply:
The opposite.  When there is no inventory_url we are not waiting for registration.
The only different here is that in case there is no inventory client we wait for node leases.  Otherwise we skip to check nodes in the cluster.

Reply:
Please elaborate with comments on why we don't wait for registration. The code isn't clear on that workflow.

Reply:
OK